README.md

```diff
@@ -1,3 +1,270 @@
+}
+
+// ConfigPath returns the path to the config file. This is particularly helpful
+// when the path is populated from environment variables or command line flags.
+// Dials will first read from environment variables and command line flags
+// and then read the config file specified by the populated field.
+func (c *Config) ConfigPath() (string, bool) {
+	// can alternatively return empty string and false if the state of the
+	// struct doesn't specify a config file to read
+	return c.Path, true
+}
+
+func main() {
+	c := &Config{}
+
+	// The following function will populate the config struct by reading the
+	// config files, environment variables, and command line flags (order matches
+	// the function name) with increasing precedence. In other words, the flag
+	// source (last) would overwrite the YAML source (first) were they both to
+	// attempt to set the same struct field. There are several options that can be
+	// passed in as well to indicate whether the file will be watched and updates
+	// to the file should update the config struct and if the flags name component
+	// separation should use different encoding
+	d, dialsErr := ez.YAMLConfigEnvFlag(context.Background(), c, WithWatchingConfigFile())
+	if dialsErr != nil {
+		// error handling
+	}
+
+	// get the struct populated from config file, environment variables, and
+	// command line flags.
+	config := &Config{}
+	d.Fill(config)
+	fmt.Printf("Config: %+v\n", config)
+}
+```
+
+For reading from JSON or TOML config files along with environment variables and command line flags,
+use the `ez.JSONConfigEnvFlag` or `ez.TOMLConfigEnvFlag` functions.
+
+If the above code is run with the following YAML file:
+
+``` yaml
+b: valueb
+val_2: 2
+val-3: false
+```
+
+and the following command (make sure to change the configpath value to point to your path)
+
+```
+export configpath=path/to/config/file
+export VAL_2=5
+go run main.go --some-val
+```
+
+the output will be 
+`Config: &{Val1:valueb Val2:5 Val3:true}`
+
+Note that even though val_2 has a value of 2 in the yaml file, the config value 
+output for Val2 is 5 because environment variables take precedence.
+
+### Configure your configuration settings
+If the predefined functions in the ez package don't meet your needs, you can specify the 
+sources you would like in the order of precedence you prefer. Not much setup is needed 
+to configure this. Choose the predefined sources and add the appropriate `dials` tags to the config struct. 
+
+``` go
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/vimeo/dials"
+	"github.com/vimeo/dials/env"
+	"github.com/vimeo/dials/file"
+	"github.com/vimeo/dials/flag"
+	"github.com/vimeo/dials/yaml"
+)
+
+type Config struct {
+	Val1 string `dials:"Val1" yaml:"b"`
+	// The `dials_env` tag is used for environment values. The Val2 value will
+	// only be populated from environment variables. If you want several different
+	// sources to be able to set this value, use the `dials` tag instead
```

Comment:
Hmm, this is misleading.
The `dials_env` and `dialsflag` tags don't prevent other sources from setting the field, they just override the name used by that source.

The second sentence should probably read something like: 
```
If you want to use a single, consistent name across several sources, set the `dials` tag instead
```